api/draft_registrations/serializers.py

```diff
@@ -0,0 +1,216 @@
+from rest_framework import serializers as ser
+from rest_framework import exceptions
+
+from framework.exceptions import PermissionsError
+from api.base.exceptions import InvalidModelValueError
+from api.base.serializers import ValuesListField, RelationshipField, LinksField, HideIfDraft, IDField
+from api.base.utils import absolute_reverse, get_user_auth
+from api.nodes.serializers import (
+    DraftRegistrationSerializerLegacy,
+    DraftRegistrationDetailLegacySerializer,
+    update_institutions,
+    get_license_details,
+    NodeLicenseSerializer,
+    NodeLicenseRelationshipField,
+    NodeContributorsSerializer,
+    NodeContributorsCreateSerializer,
+    NodeContributorDetailSerializer,
+)
+from api.taxonomies.serializers import TaxonomizableSerializerMixin
+from osf.exceptions import DraftRegistrationStateError
+from website import settings
+
+
+class NodeRelationshipField(RelationshipField):
+
+    def to_internal_value(self, node_id):
+        node = self.context['view'].get_node(node_id=node_id) if node_id else None
+        return {'branched_from': node}
+
+
+class DraftRegistrationSerializer(DraftRegistrationSerializerLegacy, TaxonomizableSerializerMixin):
+    """
+    New DraftRegistrationSerializer - instead of the node_id being provided in the URL, an optional
+    node is passed in under `branched_from`.
+
+    DraftRegistrations have several fields that can be edited that are persisted to the final registration.
+    """
+    category_choices = settings.NODE_CATEGORY_MAP.items()
+    category_choices_string = ', '.join(["'{}'".format(choice[0]) for choice in category_choices])
+
+    title = ser.CharField(required=False, allow_blank=False)
+    description = ser.CharField(required=False, allow_blank=True, allow_null=True)
+
+    category = ser.ChoiceField(required=False, choices=category_choices, help_text='Choices: ' + category_choices_string)
+    tags = ValuesListField(attr_name='name', child=ser.CharField(), required=False)
+    node_license = NodeLicenseSerializer(required=False, source='license')
+
+    links = LinksField({
+        'self': 'get_self_url',
+    })
+
+    affiliated_institutions = RelationshipField(
+        related_view='draft_registrations:draft-registration-institutions',
+        related_view_kwargs={'draft_id': '<_id>'},
+        self_view='draft_registrations:draft-registration-relationships-institutions',
+        self_view_kwargs={'draft_id': '<_id>'},
+        read_only=False,
+        many=True,
+        required=False,
+    )
+
+    branched_from = NodeRelationshipField(
+        related_view=lambda n: 'draft-nodes:draft-node-detail' if getattr(n, 'type', False) == 'osf.draftnode' else 'nodes:node-detail',
+        related_view_kwargs={'node_id': '<branched_from._id>'},
+        read_only=False,
+        required=False,
+    )
```

Comment:
if `branched_from` is a Node - this will be a node detail link.  If it's a DraftNode this will be a draftnode detail link.  DraftNode links are very sparse - and you can't access DraftNodes from node links.  DraftNode links exist solely for exposing file endpoints for uploading files to node.